debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -785,7 +787,36 @@ public synchronized Connection connection() throws SQLException {
 
     public synchronized Connection connection(boolean executeOnConnect) throws SQLException {
         if (!isConnected()) {
-            conn = factory.connect(JdbcConfiguration.adapt(config));
+
+            int maxConnectionAttempts = config.getInteger(RelationalDatabaseConnectorConfig.CONNECTION_ATTEMPTS);
+            int connectionRetryBackoff = config.getInteger(RelationalDatabaseConnectorConfig.CONNECTION_BACKOFF_MS);
+            int attempts = 0;
+            int count = 0;
+            while (attempts < maxConnectionAttempts) {
+                try {
+                    count++;
+                    LOGGER.info("Attempting to open connection #{} to {}", count, config.getString(JdbcConfiguration.HOSTNAME));
+                    conn = factory.connect(JdbcConfiguration.adapt(config));
+                    break;
+                }
+                catch (SQLException sqle) {
+                    attempts++;
+                    if (attempts < maxConnectionAttempts) {
+                        LOGGER.info("Unable to connect to database on attempt {}/{}. Will retry in {} ms.", attempts,
+                                maxConnectionAttempts, connectionRetryBackoff, sqle);
+                        try {
+                            Thread.sleep(connectionRetryBackoff);
+                        }
+                        catch (InterruptedException e) {
+                            // this is ok because just woke up early
+                        }
+                    }
+                    else {
+                        throw sqle;
+                    }
+                }
+            }
```

Comment:
Is this one actually needed in the light of the general restartability of connectors?

Reply:
This helps solve the scenario where I was failing over Aurora DB twice fast, such that the connector restart was happening during the second fail over and getting connection errors.  We could work around this by watching for the  failed task with a cronjob or something similar and restarting.

Reply:
I will remove the connection retry code for now.